prov/efa/src/efa_env.h

```diff
@@ -92,7 +95,38 @@ static inline bool efa_env_has_use_device_rdma() {
 	return (ret != -FI_ENODATA);
 }
 
-extern struct efa_env efa_env;
+/**
+ * @brief Return true if the NIC is allowed by FI_EFA_IFACE variable
+ *
+ * @param[in]	name	NIC name string
+ * @return	true if the NIC is allowed, otherwise false
+ */
+static inline bool efa_env_allows_nic(const char *name) {
+	char *match = efa_env.iface;
+	char *end = efa_env.iface + strlen(efa_env.iface);
+
+	if (name == NULL || strlen(name) < 1)
+		return false;
+
+	if (!strncmp("all", efa_env.iface, 3))
+		return true;
+
+	while (match < end) {
+		match = strstr(match, name);
+		if (!match)
+			return false;
+
+		if ((match > efa_env.iface && *(match - 1) != ',') ||
+		    ((match + strlen(name)) < end && *(match + strlen(name)) != ',')) {
+			/* Skip partial match */
+			match += strlen(name);
```

Comment:
Clarified with @wenduwan offline. But this logic is not intuitive to new readers.